### PR TITLE
Vector Chapters - Initial Commits

### PR DESCRIPTION
**Pull request by schao333**
- Added vector pages on overlays, extraction, and summary
- Uploaded shapefiles used in vector pages to _static folder
- Updated _toc.yml file to include vector pages under "Vector Operations"

### CODE REVIEW

File: pygis/docs/e_summarize_vector.md [hunk 1]
```diff
@@ -446,10 +446,7 @@ x_sp = wells["geometry"].x
 y_sp = wells["geometry"].y
 
 # Get minimum and maximum coordinate values of well points
-min_x_sp = x_sp.min()
-max_x_sp = x_sp.max()
-min_y_sp = y_sp.min()
-max_y_sp = y_sp.max()
+min_x_sp, min_y_sp, max_x_sp, max_y_sp = wells.total_bounds
```
Comment (mmann1123) [inline pygis/docs/e_summarize_vector.md:449]: ok great

File: pygis/docs/e_extraction.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+html_meta:
+  "description lang=en": "Learn how to subset and extract data from a GeoDataFrame through clipping, selecting by attribute, and selecting by location."
+  "keywords": "geospatial, attribute data, subset, extract, selection, vector, shapefile"
+  "property=og:locale": "en_US"
+---
+
+----------------
+
+```{admonition} Learning Objectives
+* Clip a vector feature using another vector feature
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:19]: This is good. Keep in mind we are trying to target key search words for SEO both here but primarily in section headings. So try to think of the way you would google it, and use that as the section title

File: pygis/docs/e_extraction.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+* Select features by their locations
+```
+```{admonition} Review
+* [Geospatial Vector Data](c_vectors.md)
+* [Attributes & Indexing for Vector Data](e_attributes.md)
+* [Creating Geospatial Vector Data](e_new_vectors.md)
+```
+----------------
+
+# Extraction
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:30]: too vague use
# Extracting spatial data

File: pygis/docs/e_extraction.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+wells = gpd.read_file("../_static/e_vector_shapefiles/sf_bay_wells_50km/sf_bay_wells_50km.shp")
+
+# Reproject data to NAD83(HARN) / California Zone 3
+# https://spatialreference.org/ref/epsg/2768/
+proj = 2768
+counties = counties.to_crs(proj)
+wells = wells.to_crs(proj)
+```
+
+We will also create a rectangle over a part of the Bay Area.
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:72]: provide a link to a platform online where you can generate (in lat lon) coordinates for a square

File: pygis/docs/e_extraction.md [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+
+    # Print title
+    print("Attribute Table: {}".format(table_name))
+
+    # Print number of rows and columns
+    print("\nTable shape (rows, columns): {}".format(attribute_table.shape))
+
+    # Display first five rows of attribute table
+    print("\nFirst five rows:")
+    display(attribute_table.head())
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:102]: `display` is a cool trick, didn't know that. however let's just show the top and bottom 2 rows so `.head(2)`. Its overwelming with top and bottom 5.

File: pygis/docs/e_extraction.md [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+# Stylize plots
+plt.style.use('bmh')
+
+# Set title
+ax.set_title('San Francisco Bay Area', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+## Clip
+
+Clip extracts and keeps only the geometries of a vector feature that are within extent of another vector feature (think of it like a cookie-cutter or mask). We can use  `clip()` in `geopandas`, with the first parameter being the vector that will be clipped and the second parameter being the vector that will define the extent of the clip. All attributes for the resulting clipped vector will be kept.
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:170]: I like the 'cookie cutter', any analogy is a good one

Comment (mmann1123) [reply pygis/docs/e_extraction.md:170]: Italize this *"All attributes for the resulting clipped vector will be kept."* its important

File: pygis/docs/e_extraction.md [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+clip_wells = gpd.clip(wells, poly)
+
+# Display attribute table
+display(clip_wells)
+
+# Plot clip
+plot_df(result_name = "San Francisco Bay Area Wells\nClip", result_df = clip_wells, result_geom_type = "point", area = poly)
+```
+
+## Select by Attribute
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:204]: ## Select Location by Attributes

File: pygis/docs/e_extraction.md [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+poly.plot(ax = ax, color = 'aquamarine', edgecolor = 'lightseagreen', alpha = 0.55)
+
+# Stylize plots
+plt.style.use('bmh')
+
+# Set title
+ax.set_title('San Francisco Bay Area', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+## Clip
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:168]: ## Clip Spatial Polygons

File: pygis/docs/e_extraction.md [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+# Select wells that are public supply
+wells_subset_1 = wells[(wells["WELL_USE"] == "Public Supply")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply", attribute_table = wells_subset_1)
+```
+
+```{code-cell} ipython3
+# Select wells that are public supply and have a depth greater than 50 ft
+wells_subset_2 = wells[(wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)]
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:229]: Great good to have a few examples of things that confuse young coders

File: pygis/docs/e_extraction.md [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft or Residential", attribute_table = wells_subset_3)
+```
+
+## Select by Location
+
+Selecting by location selects features based on its relative spatial relationship with another dataset.
+
+There are multiple spatial relationships available in `geopandas`.
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:247]: Describe using an example. Need to elaborate here a bit

[also link to arcmap description with has some drawings](https://desktop.arcgis.com/en/arcmap/10.3/map/working-with-layers/using-select-by-location.htm)

File: pygis/docs/e_extraction.md [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+# Select wells that are public supply and have a depth greater than 50 ft OR are residential
+wells_subset_3 = wells[((wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)) | (wells["WELL_USE"] == "Residential")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft or Residential", attribute_table = wells_subset_3)
+```
+
+## Select by Location
+
+Selecting by location selects features based on its relative spatial relationship with another dataset.
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:245]: maybe look at how arcmap help describes this? Could be a bit clearer

File: pygis/docs/e_extraction.md [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+| disjoint | `disjoint()` |
+| equal geometry | `geom_equals()`, `geom_almost_equals()`, `geom_equals_exact()` |
+| intersects | `intersects()` |
+| overlaps | `overlaps()` |
+| touches | `touches()` |
+| within | `within()` |
+
+Source: [GeoSeries - Binary Predicates, GeoPandas](https://geopandas.org/docs/reference/geoseries.html#binary-predicates)
+
+The functions for these spatial relationships will generally output a `pandas` series with Boolean values (`True` or `False`) whose indexing corresponds with the input dataset (from where we want to subset the data). We can use these Boolean values to subset the dataset (only the rows that have a `True` output will be retained).
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:264]: put in note admonition? unless it looks silly

File: pygis/docs/e_extraction.md [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+
+These functions can be used to select features that have the specified spatial relationship with a single Shapely vector.
+
+```{code-cell} ipython3
+# Select wells that fall within Shapely rectangle
+wells_location_subset_1 = wells[wells.within(poly_shapely)]
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:270]: use more literal names, its just good manners :)
`wells_within_poly =`

File: pygis/docs/e_extraction.md [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+
+If we're trying to select features that have a specified spatial relationship with another `geopandas` object, it gets a little tricky. This is because the `geopandas` spatial relationship functions verify the spatial relationship either row by row or index by index. In other words, the first row in the first dataset will be compared with the corresponding row or index in the second dataset, the second row in the first dataset will be compared with the corresponding row or index in the second dataset, and so on.
+
+As a result, the number of rows need to correspond or the indices numbers need to match between the two datasets--or else we'll get a warning and the output will be empty.
+
+Not easy if we want to check a bunch of features against one extent (with one geometry)... Fortunately, there are two workarounds that get the same result.
+
+Source: [geopandas.GeoSeries.within, GeoPandas](https://geopandas.org/docs/reference/api/geopandas.GeoSeries.within.html)
+
+### Workaround 1 - Manipulate extent GeoDataFrame
```
Comment (mmann1123) [inline pygis/docs/e_extraction.md:287]: this is great didn't know about this

Comment (mmann1123) [reply pygis/docs/e_extraction.md:287]: I haven't done much with these funcitons in python but I remember it being tricky and being confused on why things did and didn't work.  I do however want to make sure these are the simplest answers to this. For instance, wouldn't it also be possible to just grab the geom column value (which is stored as a shapely object i think), and comparing to that? So something like `wells[wells.within(sc_county.geometry)]`

**Review by mmann1123 — CHANGES_REQUESTED** [review]
I am leaving a few coments in extraction.md. I will look at the others later. Getting there!

File: pygis/docs/e_summarize_vector.md [hunk 15]
```diff
@@ -0,0 +1,575 @@
...
+```{admonition} Review
+* [Geospatial Vector Data](c_vectors.md)
+* [Geospatial Raster Data](c_rasters.md)
+* [Attributes & Indexing for Vector Data](e_attributes.md)
+* [Creating Geospatial Vector Data](e_new_vectors.md)
+```
+
+----------------
+
+# Summary Operations to Depict Concentrations
```
Comment (mmann1123) [inline pygis/docs/e_summarize_vector.md:32]: # Summarizing Point Patterns

**Review by mmann1123 — CHANGES_REQUESTED** [review]

File: pygis/docs/e_extraction.md (hunk 12 above)
Comment (schao333) [reply pygis/docs/e_extraction.md:264]: Looks fine to me!

File: pygis/docs/e_extraction.md (hunk 14 above)
Comment (schao333) [reply pygis/docs/e_extraction.md:287]: good point, that's an easier method

File: pygis/docs/e_extraction.md [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+import geopandas as gpd
+from shapely.geometry import Polygon
+```
+
+We will utilize shapefiles of San Francisco Bay Area county boundaries and wells within the Bay Area and the surrounding 50 km. We will first load in the data and reproject the data.
+
+```{important} All the data must have the same coordinate system in order for extraction to work correctly.
+```
+
+```{code-cell} ipython3
```
[the remark this answers is not in the document]
Comment (schao333) [reply pygis/docs/e_extraction.md:53]: I agree! I'll apply that to my chapters.

**Pull request closed without merging**